R/MeasureSurvRCLL.R

```diff
@@ -73,17 +73,53 @@ MeasureSurvRCLL = R6::R6Class("MeasureSurvRCLL",
       event = truth[, 2] == 1
       event_times = truth[event, 1]
       cens_times = truth[!event, 1]
-      distr = prediction$distr
 
-      if (!any(event)) { # all censored
-        # survival at outcome time (survived *at least* this long)
-        out[!event] = diag(as.matrix(distr[!event]$survival(cens_times)))
-      } else if (all(event)) { # all uncensored
-        # pdf at outcome time (survived *this* long)
-        out[event] = diag(as.matrix(distr[event]$pdf(event_times)))
-      } else { # mix
-        out[event] = diag(as.matrix(distr[event]$pdf(event_times)))
-        out[!event] = diag(as.matrix(distr[!event]$survival(cens_times)))
+      # Bypass distr6 construction if underlying distr represented by array
+      if (inherits(prediction$data$distr, "array")) {
+        surv = prediction$data$distr
+        if (length(dim(surv)) == 3) {
+          # survival 3d array, extract median
+          surv = .ext_surv_mat(arr = surv, which.curve = 0.5)
+        }
+        times = as.numeric(colnames(surv))
+
+        if (any(!event)) {
+          if (sum(!event) == 1) { # fix subsetting issue in case of 1 censored
```

Comment:
the `surv[!event, ]` in the case of 1 censored creates a vector, which `t` tranposes the other way than what we want

Reply:
and we need to subset here

Reply:
Really don't understand this sorry

Reply:
![image](https://github.com/mlr-org/mlr3proba/assets/10909555/d4d0f2ca-497b-4486-98b2-36086af573ab)

Reply:
and the last result is the `cdf` that is going in `C_Vec_WeightedDiscreteCdf` => which should be `times` as rows and patients as `columns` (1 in this case)

Reply:
Thanks for pointing it out - actually the code/if-condition is not complete => in the case of 1 censored but some events: `event = c(F,T,T,T)` the code takes the whole `t(1-surv)` which is wrong and I hadn't tested for this